frontend/src/components/core/Sidebar/Sidebar.tsx

```diff
@@ -101,6 +128,15 @@ class Sidebar extends PureComponent<SidebarProps, State> {
         ),
       })
     }
+
+    // Change collapse/expand preference if there are widgets on the page
+    this.setState({
+      collapsedSidebarNav: Sidebar.shouldNavCollapse(
+        this.props.appPages.length,
+        this.props.hasElements,
+        cachedSidebarNavExpandedPreference
+      ),
+    })
```

Comment:
These lines should be removed - we shouldn't have an unconditional call to `setState` in `componentDidUpdate`, but also this causes an unnecessary re-render of the component. Changes in props would already trigger a re-render, and `componentDidUpdate` is called thereafter.
Visually, it results in jumpy behavior, but is also a performance hit.